### PR TITLE
feat(next-css): Separate CSS files

### PR DESCRIPTION
**Pull request by sayuti-daniel**

### CODE REVIEW

File: packages/next-css/readme.md [hunk 1]
```diff
@@ -65,6 +65,51 @@ import "../style.css"
...
+// ./pages/_document.js
+import Document, { Head, Main, NextScript } from 'next/document'
+
+export default class MyDocument extends Document {
+  render() {
+    const { css } = this.props.buildManifest;
+    return (
+      <html>
+        <Head>
+          {css.map(href => (
```
[the remark this answers is not in the document]
Comment (sayuti-daniel) [reply packages/next-css/readme.md]: Done

**Comment by stramel** [comment]
I think combining styles per page makes sense. Currently, my initial page (login) loads all css for the app including a ton of css for draftjs which is only used on a couple of pages.

**Comment by aequasi** [comment]
Got merge conflicts @sayuti-daniel

**Comment by sayuti-daniel** [comment]
@aequasi Fixed. I haven't tested this again.

**Comment by timneutkens** [comment]
This has to be re-implemented against canary

**Comment by aequasi** [comment]
more merge conflicts.

**Comment by sayuti-daniel** [comment]
I’m still figuring out how to better implement this

**Comment by httpiago** [comment]
It would be interesting Next to have this option of separating css per page would be a great optimization of initial loading. I hope that in the future when I need this behavior this pull request has already been merged.

**Comment by codingcircus** [comment]
I tried implementing a similar feature in my nextjs app with a custom webpack config. I got the initial loading for the first page running, however I can't find a good way to load subsequent css when the route changes (on the client side). 

My current approach of loading needed css with next/head unfortunately leads to flickering while loading the next page (my guess here is that the css is only put into the head, after the component already rendered the new children). 

I think it would be best, if there was some way for `packages/next/lib/page-loader.js`to take care of loading the new css, before changing the route, similar to how the js files are currently loaded.

**Comment by aequasi** [comment]
Whats holding this back? Would love to see this.

**Comment by penx** [comment]
Does this work? Would be useful to have it published to npm under a namespace for now if so.

I'm hoping to implement path splitting for CSS modules (loaded via npm) in this project - https://github.com/penx/govuk-frontend-react-example-next would be great if I can do so via this.

Also would like to hear what further work is needed on this and if there's anything I could help with.

**Comment by resir014** [comment]
Any updates on this? This would be really nice to have.

**Comment by skywickenden** [comment]
Also much needed. I have a large site in next.js and the css loaded on every page is now 730k! Should be about 30k per page.

**Comment by timneutkens** [comment]
Looks like this was auto-closed when removing the canary branch. We're currently discussing how to solve this issue in a different way, I've sent you a dm on spectrum @sayuti-daniel

**Comment by penx** [comment]
@timneutkens please could you reopen or put a status update here for the rest of us?

**Comment by penx** [comment]
or raise a new issue (don't think there's an existing one)?

**Comment by jagreehal** [comment]
@timneutkens any updates?

**Comment by 2012mjm** [comment]
Does it work for SASS ?

**Comment by penx** [comment]
This will hopefully be covered by https://github.com/zeit/next.js/issues/8626

**Pull request closed without merging**